pkg/build/controller/controller.go

```diff
@@ -102,9 +102,13 @@ func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
 	}
 
 	if buildutil.IsBuildComplete(build) {
+		if _, ok := build.Annotations[buildapi.BuildCompletedAnnotation]; ok {
+			return nil
+		}
```

Comment:
also memory caches die. we need this persisted so we don't have to run through all the builds again on a restart of the controller.

The distinction is a complete build means the build is complete.  this annotation means "and we don't care about it anymore"  (we still care about complete builds, the first time we see them enter the complete state)